src/crc32.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2024 Space Cubics, LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/ztest.h>
+#include <csp/csp.h>
+#include <csp/csp_crc32.h>
+#include <csp/csp_id.h>
+#include <endian.h>
+
+struct crc_fixture {
+	csp_packet_t *buf;
+};
+
+static struct crc_fixture crc_fixture;
+
+static void *setup(void)
+{
+	csp_init();
+
+	return &crc_fixture;
+}
+
+static void before(void *data)
+{
+	struct crc_fixture *fixture = data;
+
+	fixture->buf = csp_buffer_get(0);
+	zassert_true(fixture->buf != NULL, NULL);
+}
+
+static void after(void *data)
+{
+	struct crc_fixture *fixture = data;
+
+	csp_buffer_free(fixture->buf);
+}
+
+ZTEST_F(crc, test_crc_verify_without_crc)
+{
+	csp_packet_t *packet = fixture->buf;
+	packet->length = sizeof(uint32_t) - 1;
```

Comment:
Why set the `sizeof(uint32_t) - 1 ` ?
I think this setting is inconsistent with the test name `without crc`.

Reply:
I was trying to test here : https://github.com/libcsp/libcsp/blob/develop/src/csp_crc32.c#L116-L118
But now I see that I'm not trying any more to test here : https://github.com/libcsp/libcsp/blob/develop/src/csp_crc32.c#L132-L134
Should I test the first error with the size part ? If not I will just go back to the previous code.

Reply:
I understand your purpose, but I think:

If you set the `sizeof(uint32_t) - 1`, you need to set the 3 byte data to `packet->data`.
Certainly, there is no point in setting data since it will be rejected by the length check, but I think the test code should comply with the original usage as much as possible.

And, one more things.

 If we executes `csp_crc32_verify()` without `csp_crc32_append()`, as you say, there are two error routes.
  1. packet length less than 4 byte ( https://github.com/libcsp/libcsp/blob/develop/src/csp_crc32.c#L116-L118 )
  2. CRC value is unmatched (  https://github.com/libcsp/libcsp/blob/develop/src/csp_crc32.c#L132-L134 )
 
So in this test, we should be test the both case like below.
(It's just in example)

```
    csp_packet_t *packet = fixture->buf;
    packet->length = 0;

    ret = csp_crc32_verify(packet);
    zassert_equal(ret, CSP_ERR_CRC32, "Error on csp_crc32_verify");↲

    memcpy(packet->data, "Hello", 5);
    packet->length = 5;
    ret = csp_crc32_verify(packet);
    zassert_equal(ret, CSP_ERR_CRC32, "Error on csp_crc32_verify");↲
```

Anyway let's discuss later in F2F.